### PR TITLE
Fixes #25600 - Migrate Icon to patternfly-react

### PR DESCRIPTION
**Pull request by orrabin**
Remove Icon from foreman
Switch DocumentLinkContent to use patternfly-react Icon

### CODE REVIEW

**Review by amirfefer — DISMISSED** [review]
Thanks @orrabin !  
Works well :+1: 
I just wonder why [patternfly design](https://www.patternfly.org/pattern-library/communication/empty-state/#design) lacks with this question mark icon? how about open an issue/suggestion to patternfly react?

**Review by sharvit — APPROVED** [review]
LGTM! Thanks @orrabin :+1:

**Comment by sharvit** [comment]
> I just wonder why patternfly design lacks with this question mark icon? how about open an issue/suggestion to patternfly react?

@amirfefer I don't think it is about lack of design. I guess we just decided we want a button that link to the documentations and our `documentation-button` uses `question-sign`.

**Comment by amirfefer** [comment]
@sharvit, as we discussed it earlier, the documentation button (in empty state component) is a link, not a button. so I would remove the question mark icon from it.

File: webpack/assets/javascripts/react_app/components/common/EmptyState/__snapshots__/EmptyState.test.js.snap [hunk 1]
```diff
@@ -58,7 +58,8 @@ exports[`Default Empty State should render documentation when given a url 1`] =
       target="_blank"
     >
       <span
```
Comment (sharvit) [inline webpack/assets/javascripts/react_app/components/common/EmptyState/__snapshots__/EmptyState.test.js.snap]: As we discussed, there is no reason to have Icon here.
@orrabin can remove the Icon from the `EmptyState` please?

**Review by sharvit — CHANGES_REQUESTED** [review]

File: webpack/assets/javascripts/react_app/components/common/DocumentationLink/index.js [hunk 2]
```diff
@@ -1,13 +1,12 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { MenuItem } from 'patternfly-react';
-import Icon from '../Icon';
```
Comment (waldenraines) [inline webpack/assets/javascripts/react_app/components/common/DocumentationLink/index.js:4]: Can we remove the Icon component itself?

**Review by waldenraines — COMMENTED** [review]
Looking good so far, thanks @orrabin!

File: webpack/assets/javascripts/react_app/components/common/DocumentationLink/index.js (hunk 2 above)
Comment (orrabin) [reply webpack/assets/javascripts/react_app/components/common/DocumentationLink/index.js:4]: I removed the Icon folder. What else should be removed?

**Review by sharvit — DISMISSED** [review]
LGTM! Thanks @orrabin :+1:

File: webpack/assets/javascripts/react_app/components/common/DocumentationLink/index.js [hunk 3]
```diff
@@ -1,13 +1,12 @@
...
+import { MenuItem, Icon } from 'patternfly-react';
 import { newWindowOnClick } from '../../../common/helpers';
 import { translate as __ } from '../../../../react_app/common/I18n';
 
 export const DocumentLinkContent = ({ children }) => (
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/components/common/DocumentationLink/index.js]: Do we need this component?

**Review by amirfefer — CHANGES_REQUESTED** [review]
Thanks @orrabin :+1: 
Can we delete`DocumentationLink` component after we removed it from its consumer?

**Comment by ohadlevy** [comment]
@orrabin whats the status of the PR? thanks

**Comment by orrabin** [comment]
@ohadlevy updated today resolving the comment from @amirfefer

**Review by sharvit — APPROVED** [review]

**Review by ohadlevy — APPROVED** [review]

**Pull request merged**